org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/XMLFormatterDocumentNew.java

```diff
@@ -519,7 +519,7 @@ void replaceSpacesWithOneSpace(int spaceStart, int spaceEnd, List<TextEdit> edit
 	 */
 	public FormatElementCategory getFormatElementCategory(DOMElement element,
 			XMLFormattingConstraints parentConstraints) {
-		if (!element.isClosed()) {
+		if (!element.isClosed() && parentConstraints != null) {

```

Comment:
Could you add a comment with a sample where parentConstraints  is null.

Reply:
I think this is no longer needed since the tests pass without it now and I have removed it